File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string ans = "";
        int i = a.size()-1;
        int j = b.size()-1;
        int temp = 0;
        while(i>=0 || j>=0 || temp==1) {
            temp += i>=0 ? a[i]-'0' : 0;
            temp += j>=0 ? b[j]-'0' : 0;
            ans = char(temp%2 + '0') + ans;
            temp /= 2;
            i--; j--;
        }
        return ans;
    }
};

File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        String ans = "";
        int carry = 0;
        int i = a.length() - 1;
        int j = b.length() - 1;
        while(i >= 0 || j >= 0 || carry == 1) {
            int a1 = i >= 0 ? a.charAt(i) - '0' : 0;
            int b1 = j >= 0 ? b.charAt(j) - '0' : 0;
            int sum = a1 + b1 + carry;
            ans = Integer.toString(sum % 2) + ans;
            carry = sum / 2;
            i--; j--;
        }
        return ans;
    }
}